### PR TITLE
EWPP-992: Update contacts rendering on full and teaser view displays.

### PR DESCRIPTION
**Pull request by 22Alexandra**

### CODE REVIEW

File: modules/oe_theme_content_organisation/src/Plugin/ExtraField/Display/TeaserDetailsExtraField.php [hunk 1]
```diff
@@ -85,32 +79,38 @@ public function viewElements(ContentEntityInterface $entity) {
...
+      $cache->addCacheableDependency($contact);
+      $cache->addCacheableDependency($contact_access);
 
-    if (!$contact_access->isAllowed()) {
-      $cache->applyTo($build);
-      return $build;
-    }
+      if (!$contact_access->isAllowed()) {
+        $cache->applyTo($build);
+        return $build;
```
Comment (upchuk) [inline modules/oe_theme_content_organisation/src/Plugin/ExtraField/Display/TeaserDetailsExtraField.php]: So if the first contact is not accessible, we return and don't continue the loop?

File: modules/oe_theme_content_organisation/src/Plugin/ExtraField/Display/TeaserDetailsExtraField.php [hunk 2]
```diff
@@ -85,32 +79,38 @@ public function viewElements(ContentEntityInterface $entity) {
...
+        'oe_email' => ['type' => 'email_mailto'],
+        'oe_phone' => [],
+        'oe_address' => [
+          'type' => 'oe_theme_helper_address_inline',
+          'settings' => ['delimiter' => ', '],
+        ],
+      ];
+      foreach ($fields as $field_name => $display_options) {
+        if (!$contact->get($field_name)->isEmpty()) {
+          $items[] = $this->getRenderableFieldListItem($contact, $field_name, $display_options);
```
Comment (upchuk) [inline modules/oe_theme_content_organisation/src/Plugin/ExtraField/Display/TeaserDetailsExtraField.php:108]: You are merging here the fields of all the contacts and this is not what needs to happen if you look at the screenshot. 

I have two contacts:

Screen Shot 2021-04-20 at 09 28 53

And this is the resulting teaser:

Screen Shot 2021-04-20 at 09 29 19

Compare please to the screenshot from the ticket.

**Review by upchuk — CHANGES_REQUESTED** [review]

File: modules/oe_theme_content_organisation/src/Plugin/ExtraField/Display/TeaserDetailsExtraField.php [hunk 3]
```diff
@@ -71,46 +70,47 @@ public function viewElements(ContentEntityInterface $entity) {
...
-    ];
+    $build = [];
     $cache = CacheableMetadata::createFromRenderArray($build);
 
-    $contact_access = $contact->access('view', NULL, TRUE);
-    $cache->addCacheableDependency($contact);
-    $cache->addCacheableDependency($contact_access);
-
-    if (!$contact_access->isAllowed()) {
-      $cache->applyTo($build);
```
Comment (upchuk) [inline modules/oe_theme_content_organisation/src/Plugin/ExtraField/Display/TeaserDetailsExtraField.php:94]: Why are you applying the cache to the build here alreeady?

File: modules/oe_theme_content_organisation/src/Plugin/ExtraField/Display/TeaserDetailsExtraField.php [hunk 4]
```diff
@@ -71,46 +70,47 @@ public function viewElements(ContentEntityInterface $entity) {
...
-      return $build;
-    }
+    /** @var \Drupal\oe_content_entity_contact\Entity\ContactInterface $contact */
+    foreach ($entity->get('oe_organisation_contact')->referencedEntities() as $key => $contact) {
```
Comment (upchuk) [inline modules/oe_theme_content_organisation/src/Plugin/ExtraField/Display/TeaserDetailsExtraField.php]: as $delta

**Review by upchuk — CHANGES_REQUESTED** [review]

**Review by upchuk — APPROVED** [review]

File: modules/oe_theme_content_organisation/src/Plugin/ExtraField/Display/TeaserDetailsExtraField.php [hunk 5]
```diff
@@ -71,46 +70,46 @@ public function viewElements(ContentEntityInterface $entity) {
...
-    $cache->addCacheableDependency($contact);
-    $cache->addCacheableDependency($contact_access);
-
-    if (!$contact_access->isAllowed()) {
-      $cache->applyTo($build);
-      return $build;
-    }
+    /** @var \Drupal\oe_content_entity_contact\Entity\ContactInterface $contact */
+    foreach ($entity->get('oe_organisation_contact')->referencedEntities() as $delta => $contact) {
+      if (!$contact) {
```
Comment (sergepavle) [inline modules/oe_theme_content_organisation/src/Plugin/ExtraField/Display/TeaserDetailsExtraField.php:78]:
```suggestion
      if (!$contact instanceof ContactInterface) {
```

**Review by sergepavle — APPROVED** [review]

**Review by sergepavle — APPROVED** [review]

**Pull request merged**